tools/gen_git_sha.sh

```diff
@@ -1,8 +1,13 @@
 #!/bin/bash
 
-SHA=`git -C $1 rev-parse HEAD`
-grep -q $SHA $2
-if [[ $? -ne 0 ]]; then
-  echo -e "#include \"common/common/version.h\"\n\n" > $2
-  echo -e "const std::string VersionInfo::GIT_SHA(\"$SHA\");" >> $2
-fi
+[[ -z "${GIT_BUILD_SHA}" ]] && GIT_BUILD_SHA="0000000000000000000000000000000000000000"
+
+cat <<EOF > version_generated.cc
+#include "common/common/version.h"
+
+const std::string VersionInfo::GIT_SHA("${GIT_BUILD_SHA}");
+EOF
+
+cat <<EOF > BUILD
+exports_files(["version_generated.cc"])
```

Comment:
If you create a `cc_library` here instead of using `exports_files`, you will be able to avoid the warning, referring to it as `@git_version//:version_generated` instead.